crates/provider-http-server/src/address.rs

```diff
@@ -80,19 +94,45 @@ impl Provider for HttpServerProvider {
             }
         };
 
-        // Start a server instance that calls the given component
-        let http_server = HttpServerCore::new(Arc::new(settings), link_config.target_id)
-            .await
-            .context("httpserver failed to start listener for component")?;
-
-        // Save the component and server instance locally
-        self.component_handlers.write().await.insert(
-            (
-                link_config.target_id.to_string(),
-                link_config.link_name.to_string(),
-            ),
-            http_server,
+        let component_meta = (
+            link_config.target_id.to_string(),
+            link_config.link_name.to_string(),
         );
+        let mut sockets_by_link_name = self.sockets_by_link_name.write().await;
+        let mut handlers_by_socket = self.handlers_by_socket.write().await;
+
+        match sockets_by_link_name.entry(link_config.link_name.to_string()) {
+            // If a mapping already exists, and the stored address is different, disallow overwriting
+            std::collections::hash_map::Entry::Occupied(_) => {
+                bail!("overwriting links is not currently supported")
```

Comment:
Thoughts on noting why this was a conflict, e.g. a different component already has the address v.s. this component has a different address?

Reply:
Yeah will add something to this to make it a little clearer